src/server/editorServices.ts

```diff
@@ -744,18 +825,58 @@ module ts.server {
                 this.psLogger.info(project.filesToString());
                 this.psLogger.info("-----------------------------------------------");
             }
-            this.psLogger.info("Open file roots: ")
+            for (var i = 0, len = this.configuredProjects.length; i < len; i++) {
+                var project = this.configuredProjects[i];
+                project.updateGraph();
+                this.psLogger.info("Project (configured) " + (i+this.inferredProjects.length).toString());
+                this.psLogger.info(project.filesToString());
+                this.psLogger.info("-----------------------------------------------");
+            }
+            this.psLogger.info("Open file roots of inferred projects: ")
             for (var i = 0, len = this.openFileRoots.length; i < len; i++) {
                 this.psLogger.info(this.openFileRoots[i].fileName);
             }
-            this.psLogger.info("Open files referenced: ")
+            this.psLogger.info("Open files referenced by inferred or configured projects: ")
             for (var i = 0, len = this.openFilesReferenced.length; i < len; i++) {
-                this.psLogger.info(this.openFilesReferenced[i].fileName);
+                var fileInfo = this.openFilesReferenced[i].fileName;
+                if (this.openFilesReferenced[i].defaultProject.isConfiguredProject()) {
+                    fileInfo += " (configured)";
+                }
+                this.psLogger.info(fileInfo);
+            }
+            this.psLogger.info("Open file roots of configured projects: ")
+            for (var i = 0, len = this.openFileRootsConfigured.length; i < len; i++) {
+                this.psLogger.info(this.openFileRootsConfigured[i].fileName);
             }
             this.psLogger.endGroup();
         }
 
-        openConfigFile(configFilename: string): ProjectOpenResult {
+        configProjectIsActive(fileName: string) {
+            for (var i = 0, len = this.configuredProjects.length; i < len; i++) {
+                if (this.configuredProjects[i].projectFilename == fileName) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        findConfigFile(openedFileName: string): string {

```

Comment:
this looks almost exactly what tsc.ts/findConfigFile does, except that this takes a directory as an argument. i would move the other function to utilities.ts, or program.ts and use it, unless there is a reason for the duplication.